seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSource.java

```diff
@@ -57,13 +60,22 @@ public Boundedness getBoundedness() {
 
     @Override
     public void prepare(Config pluginConfig) throws PrepareFailException {
-        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, HttpConfig.URL);
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, HttpConfig.URL, HttpConfig.SCHEMA);
         if (!result.isSuccess()) {
             throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
         }
         this.httpParameter.buildWithConfig(pluginConfig);
-        // TODO support user custom row type
-        this.rowType = new SeaTunnelRowType(new String[]{"content"}, new SeaTunnelDataType<?>[]{BasicType.STRING_TYPE});
+        Config schema = pluginConfig.getConfig(HttpConfig.SCHEMA);
+        this.rowType = SeatunnelSchema.buildWithConfig(schema).getSeaTunnelRowType();
+        String format;
```

Comment:
Seem like your remove text format, please add it back, so it can support json and text both.

Reply:
Resolved, please review.